### DISCUSSION TITLE
MicronautTest trying to load classes in src/test of 3rd party dependencies #707

### DISCUSSION THREAD

**Discussion by subash89** [micronaut-projects/micronaut-core#8668] [General] [answered]
In my micronaut project-A pom.xml, I have defined project-B as a dependency.
Only when try to run a MicronautTest , It tries to load classes in package "com.myapp.junit5.aws" which is a class in project-B/src/test folder. Why micronaut trying to load those test package classes ? The app itself not complaining about those, but only the test.

`@MicronautTest
public class SimpleTest {

        @Test
        void testStartEngine() {
            System.out.println("success");
        }
}
`

`Caused by: java.lang.NoClassDefFoundError: **software/amazon/awssdk/services/firehose/FirehoseAsyncClient** at com.myapp.junit5.aws.$FirehoseTestResource$Definition.<clinit>(Unknown Source) at com.myapp.junit5.aws.$FirehoseTestResource$Definition$Reference.load(Unknown Source) at io.micronaut.context.AbstractInitializableBeanDefinitionReference.load(AbstractInitializableBeanDefinitionReference.java:145) at io.micronaut.context.DefaultBeanContext.lambda$getBeanDefinitions$17(DefaultBeanContext.java:1503) at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195) at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177) at java.base/java.util.concurrent.ConcurrentLinkedQueue.forEachFrom(ConcurrentLinkedQueue.java:1037) at java.base/java.util.concurrent.ConcurrentLinkedQueue$CLQSpliterator.forEachRemaining(ConcurrentLinkedQueue.java:894) at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484) at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474) at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913) at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578) at io.micronaut.context.DefaultBeanContext.getBeanDefinitions(DefaultBeanContext.java:1506) at io.micronaut.context.AnnotationProcessorListener.onCreated(AnnotationProcessorListener.java:64) at io.micronaut.context.AnnotationProcessorListener.onCreated(AnnotationProcessorListener.java:44) at io.micronaut.context.DefaultBeanContext.triggerBeanCreatedEventListener(DefaultBeanContext.java:2407) at io.micronaut.context.DefaultBeanContext.postBeanCreated(DefaultBeanContext.java:2385) at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2309) at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2251) at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3016) at io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:80) at io.micronaut.context.DefaultBeanContext.findOrCreateSingletonBeanRegistration(DefaultBeanContext.java:2918)`

**Comment by graemerocher** [comment, marked as answer]
perhaps provide an example that reproduces the issue in an issue report

**Reply by subash89** [reply]
When I try to replicate the issue into a much simpler example project to recreate, figured out micronaut didn't like provided scope dependencies of that 3rd party library. It try to load those if they are in provided scope. 
When I remove provided scope of those dependencies defined in the 3rd party module, then micronautTest didn't try to load them.
